libvncclient/rfbproto.c

```diff
@@ -1780,6 +1850,15 @@ SendClientCutText(rfbClient* client, char *str, int len)
 
   if (!SupportsClient2Server(client, rfbClientCutText)) return TRUE;
 
+  if ((!str) || (len < 0)) {
+    rfbClientLog("SendClientCutText: str %p, len %d\n", str, len);
+    return FALSE;
+  }
+
+  if (client->appData.clipboardCap) { /* if support extended clipboard, use it */
```

Comment:
Please check via SupportsClient2Server()

Reply:
it is checked before. code here is to chose use old style clipboard or new style clipboard(under the pseudo encoding but reuse the same message)